internal/ingest/linksystem.go

```diff
@@ -275,6 +232,47 @@ func (ing *Ingester) ingestAd(publisherID peer.ID, adCid cid.Cid, ad schema.Adve
 	return nil
 }
 
+// ingestEntryChunk ingests a block of entries as that block is received
+// through graphsync.
+//
+// When each advertisement on a chain is processed by ingestAd, that
+// advertisement's entries are synced in a spearate legs.Subscriber.Sync
+// operation. This function is used as a scoped block hook, and is called for
+// each block that is received.
+func (ing *Ingester) ingestEntryChunk(publisher peer.ID, adCid cid.Cid, ad schema.Advertisement, entryChunkCid cid.Cid) error {
+	log := log.With("publisher", publisher, "adCid", adCid, "cid", entryChunkCid)
+
+	// Get data corresponding to the block.
+	entryChunkKey := dsKey(entryChunkCid.String())
+	val, err := ing.ds.Get(context.Background(), entryChunkKey)
```

Comment:
worth having a context for this function, rather than using context.Background throughout?

Reply:
done